Cython/Utils.py

```diff
@@ -264,15 +265,24 @@ def check_package_dir(dir_path, package_names):
 @cached_function
 def contains_init(dir_path):
     for filename in PACKAGE_FILES:
```

Comment:
Running `glob()` many times can be fairly expensive. Let's not go above the number of files in the directory.
```suggestion
    return any(_is_package_filename(filename) for filename in os.listdir(dir_path)):
```